yasnippet.el

```diff
@@ -3035,18 +3035,30 @@ through the field's start point"
    (not (and (yas--field-number field)
              (zerop (yas--field-number field))))))
 
-(defun yas--snippets-at-point (&optional all-snippets)
-  "Return a sorted list of snippets at point.
-
-The most recently-inserted snippets are returned first."
-  (sort
-   (delq nil (delete-dups
-              (mapcar (lambda (ov) (overlay-get ov 'yas--snippet))
-                      (if all-snippets (overlays-in (point-min) (point-max))
-                        (nconc (overlays-at (point))
-                               (overlays-at (1- (point))))))))
-   #'(lambda (s1 s2)
-       (<= (yas--snippet-id s2) (yas--snippet-id s1)))))
+(defun yas-active-snippets (&optional where)
+  "Return a sorted list of active snippets.
+The most recently-inserted snippets are returned first.
+
+By default, lists snippets around point.  If optional argument
```

Comment:
> You mean (defun yas-active-snippets (&optional begin end) instead of requiring consing?

Yes

> I'm not sure if the docstring should really be committing to a particular implementation.

That's a valid concern, but given that active snippets should and do behave like overlays, per design, we should spare ourselves the maintenance burden of keeping an independent description of that behaviour. Therefore I try to defer to an admittedly more stable API, in this case Emacs'. So even if we implement snippets in terms of some other kind of primitive, they will still behave like overlays, and we keep honouring the docstring. Many of Emacs's own functions use this type of delegation, you find it mostly when the docstring says "<function-name>, which see".

But as I said, I'm not opposed to a new protocol with yasnippet-specific semantics if you feel its worth it.